src/erlzk.erl

```diff
@@ -67,9 +67,14 @@ connect(ServerList, Timeout) ->
 
 %% @doc Connect to ZooKeeper.
 %% @see connect/4
--spec connect(server_list(), pos_integer(), options()) -> {ok, pid()} | {error, atom()};
-             ({local, Name::atom()} | {global, GlobalName::term()} | {via, Module::atom(), ViaName::term()},
-              server_list(), pos_integer()) -> {ok, pid()} | {error, atom()}.
+-spec connect(ServerList :: server_list(), 
+              Timeout :: pos_integer(), 
+              Options :: options()) -> supervisor:startchild_ret();
+             (ServerName :: {local, Name::atom()} | 
+                            {global, GlobalName::term()} | 
+                            {via, Module::atom(), ViaName::term()},
+              ServerList :: server_list(), 
+              Timeout :: pos_integer()) -> supervisor:startchild_ret().
```

Comment:
Diff turned out to be a bit difficult to read here, but the only change to the spec is the change of return type from `{ok, pid()} | {error, atom()}` to `supervisor:startchild_ret()`. Same goes for `connect/3` below.